File: enJAVA/funciones.java
import java.lang.Math;

class Funciones {
    public static void main(String[] args) {
        System.out.println(repeat("hola",3));
        
    }

    public static int ping (String ip){
        return 0;
    }

    public static double pmt (double tasaI, int numPeriodos, double pv){ //done
        double resul = (pv*tasaI)/(1 - (Math.pow((1 + tasaI), -numPeriodos)));
        int res = (int)(resul);
        return res;
    }

    public static int bin2dec (String binary){
        int dec = Integer.parseInt(binary, 2);
        return dec;
    }

    public static String dec2bin (int valDec){ //done
        StringBuilder binary = new StringBuilder();
        int i =0;
        while (valDec > 0){
            binary.append(valDec%2);
            i++;
            valDec = valDec/2;
        }
        return binary.reverse().toString();
    }

    public static int c2f (int tempC){ //done
        int tempF = (int)(tempC * 1.8) + 32;
        return tempF;
    }

    public static int f2c (int tempF){ //done
        int tempC = (int)((tempF-32) *  0.5556);
        return tempC;
    }

    public static int factorial (int num){ //done
        int i,fact=1;  
        for(i=1;i<=num;i++){    
            fact=fact*i;    
        }    
        return fact;
    }

    public static int hex2dec (String hexa){
        int decimal = Integer.parseInt(hexa,16);  
        return decimal;
    }

    public static String dec2hex (int num){
        String hexa = Integer.toHexString(num);
        return hexa;
    }

    public static int compareString (String str1, String str2){ //done
        if(str1.length() < str2.length()){
            return -1;
        } else if(str1.length() == str2.length()){
            if(str1 == str2){
                return 0;
            } else {
                return -1;
            }
        } else if(str1.length() > str2.length()){
            return 1;
        }
        return -2;
    }

    public static String trim (String cadena, String remover){ //done
        if(cadena.charAt(0) == remover.charAt(0)){
            StringBuilder sb = new StringBuilder(cadena);
            String n = sb.deleteCharAt(0).toString();
            return n;
        } else if(cadena.charAt(cadena.length()-1) == remover.charAt(0)){
            StringBuilder sb = new StringBuilder(cadena);
            String n = sb.deleteCharAt(cadena.length()-1).toString();
            return n;
        }
        return ""; //no existen los caracteres 
    }

    public static String repeat (String str, int cant){
        String res = "";
        for(int a = 0; a<cant; a++){
            res+=str;
        }
        return res;
    }



}

File: enCPP/funciones.cpp
#include <stdio.h>
#include <string>
#include <sstream>
#include <tgmath.h>
#include <iostream>
#include <iomanip>
#include <stdlib.h>

using namespace std;

struct Funciones{
public:
    Funciones(){};
    int ping(string ip){
        return 0;
    }
    double pmt (double tasaI, int numPeriodos, double pv){ //done
        double resul = (pv*tasaI)/(1 - (pow((1 + tasaI), -numPeriodos)));
        int res = (int)(resul);
        return res;
    }

    int bin2dec (string binary){
        int n = stoi(binary);
        int dec = 0, i = 0, remainder;
        while (n!=0){
            remainder = n%10;
            n /= 10;
            dec += remainder*pow(2,i);
            ++i;
        }
        return dec;
    }

    string dec2bin (int valDec){ //done
        int binary = 0;
        int remainder, i = 1;

        while (valDec!=0){
            remainder = valDec%2;
            valDec /= 2;
            binary += remainder*i;
            i *= 10;
        }
        return to_string(binary);
    }

    int c2f (int tempC){ //done
        int tempF = (int)(tempC * 1.8) + 32;
        return tempF;
    }

    int f2c (int tempF){ //done
        int tempC = (int)((tempF-32) *  0.5556);
        return tempC;
    }

    int factorial (int num){ //done
        int i,fact=1;  
        for(i=1;i<=num;i++){    
            fact=fact*i;    
        }    
        return fact;
    }

    int hex2dec (string hexa){
        int dec;
        stringstream ss;
        ss << hex << hexa;
        ss >> dec;
        return dec;
    }

    string dec2hex (int num){
        stringstream ss;
        ss << hex << num;
        string res = ss.str();
        transform(res.begin(), res.end(),res.begin(), ::toupper);
        return res;
    }

    int compareString (string str1, string str2){ //done
        if(str1.length() < str2.length()){
            return -1;
        } else if(str1.length() == str2.length()){
            if(str1 == str2){
                return 0;
            } else {
                return -1;
            }
        } else if(str1.length() > str2.length()){
            return 1;
        }
        return -2;
    }

    string trim (string cadena, string remover){ //done
        if(cadena[0] == remover[0]){
            cadena.erase(remove(cadena.begin(), cadena.end(), cadena[0]), cadena.end());
            return cadena;
        } else if(cadena[cadena.length()-1] == remover[0]){
            cadena.erase(remove(cadena.begin(), cadena.end(), cadena[cadena.length()-1]), cadena.end());
            return cadena;
        }
        return ""; //no existen los caracteres 
    }

    string repeat (string str, int cant){
        string res = "";
        for(int a = 0; a<cant; a++){
            res+=str;
        }
        return res;
    }

};

int main(){

    Funciones * funciones = new Funciones();
    cout<<funciones->bin2dec("1001")<<endl;
    cout<<funciones->dec2bin(9)<<endl;
    return 0;
}
